stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -856,6 +959,24 @@ public func _graph<State : _TensorArrayProtocolEnhanced,
   }
 }
 
+
+public func _tffunc<State : _TensorArrayProtocolEnhanced,
+                    Data : TensorGroup>(
+  with state: State,
+  in fn: @escaping (State, Data) -> (State)
+) -> (Data) -> (String) {
+  let wrappedFn = {
+    // The result argument needs to be a type that conforms to TensorGroup.
+    // We are arbitrarily picking Tensor<Float> here.
+    (s: State, d: Data) -> (State, Tensor<Float>?) in
+      (fn(s, d), nil)
+  }
+  let traceContext = _trace(with: state, in : wrappedFn)
+  return { (data: Data) -> String in
+    traceContext.specializedTFFunc(data: data)
```

Comment:
IIUC, when we apply this design to creating a data input pipeline function, this will generate 1 new TF function for each data batch, and could be an efficiency concern?

Ok to submit this patch to unblock development, but we should follow up on this issue (e.g. create a  unit test to show how tracing can be used to create an input pipeline function).

Reply:
Yes, this will return new function every time this is called. That is a valid concern. We could cache the functions keyed by data to avoid creating new functions? 

 I am working on the data pipeline right now.